File: Android/FinalProjectZhuZihan/app/src/main/java/itp341/zhu/zihan/finalprojectzhuzihan/Model/Country.java
package itp341.zhu.zihan.finalprojectzhuzihan.Model;

public class Country {
    public String Name,FlagUrl,Population,Language,TimeZone,Area,Currency,Capital;

    public Country(){
        Name ="";
        FlagUrl = "";
        Population = "";
        Language = "";
        TimeZone = "";
        Area ="";
        Currency = "";
        Capital = "";
    }

    public Country(String name, String flagUrl, String Population,String Language, String timeZone, String area, String currency, String capital) {
        Name = name;
        FlagUrl = flagUrl;
        this.Population = Population;
        this.Language = Language;
        TimeZone = timeZone;
        Area = area;
        Currency = currency;
        Capital = capital;
    }

    @Override
    public String toString() {
        return "CountryCardView{" +
                "Name='" + Name + '\'' +
                ", FlagUrl='" + FlagUrl + '\'' +
                ", Population='" + Population + '\'' +
                ", Language='" + Language + '\'' +
                ", TimeZone='" + TimeZone + '\'' +
                ", Area='" + Area + '\'' +
                ", Currency='" + Currency + '\'' +
                ", Capital='" + Capital + '\'' +
                '}';
    }

    public String getName() {
        return Name;
    }

    public String getFlagUrl() {
        return FlagUrl;
    }

    public String getPopulation() {
        return Population;
    }

    public String getLanguage() {
        return Language;
    }

    public String getTimeZone() {
        return TimeZone;
    }

    public String getArea() {
        return Area;
    }

    public String getCurrency() {
        return Currency;
    }

    public String getCapital() {
        return Capital;
    }

    public void setName(String name) {
        Name = name;
    }

    public void setFlagUrl(String flagUrl) {
        FlagUrl = flagUrl;
    }

    public void setPopulation(String population) {
        Population = population;
    }

    public void setLanguage(String language) {
        Language = language;
    }

    public void setTimeZone(String timeZone) {
        TimeZone = timeZone;
    }

    public void setArea(String area) {
        Area = area;
    }

    public void setCurrency(String currency) {
        Currency = currency;
    }

    public void setCapital(String capital) {
        Capital = capital;
    }
}

File: Swift/TravelHelper/TravelHelper/TravelHelper/Model/Country.swift
//
//  Country.swift
//  TravelHelper
//
//  Created by Zihan Zhu on 12/9/19.
//  Copyright © 2019 Zihan Zhu. All rights reserved.
//

import Foundation

struct Country {
    //The model for the info of a country
    private var Name : String
    private var FlagUrl : String//the url of national flag
    private var Population : String
    private var Language : String
    private var TimeZone : String
    private var Area : String
    private var Currency : String
    private var Capital : String
    init(name:String,flagurl:String,population:String,language:String,timezone:String,area:String,currency:String,capital:String){
        self.Name = name
        self.FlagUrl = flagurl
        self.Population = population
        self.Language = language
        self.TimeZone = timezone
        self.Area = area
        self.Currency = currency
        self.Capital = capital
    }
    //getters
    public func getName()->String{
        return Name
    }
    
    public func getFlagUrl()->String{
        return FlagUrl
    }
    
    public func getPopulation()->String{
        return Population
    }
    
    public func getLanguage()->String{
        return Language
    }
    
    public func getTimezone()->String{
        return TimeZone
    }
    
    public func getArea()->String{
        return Area
    }
    
    public func getCurrency()->String{
        return Currency
    }
    
    public func getCapital()->String{
        return Capital
    }
}
